tests/features/user-management.feature

```diff
@@ -0,0 +1,21 @@
+@api @javascript
+Feature: Manage users
+  In order to manage the users of the site
+  As an administrator with the appropriate rights
+  I need to be able to edit and delete users
+
+  Scenario: A user with the appropriate rights can delete users.
+    Given the site is configured to use Drupal login
+    And I am logged in as a user with the "administer users, delete user" permissions
```

Comment:
It looked to me that the test doesn't cover enough, so I went and tried do delete a user through the bulk action "Cancel the selected user accounts". 
That form `UserMultipleCancelConfirm` is indeed not covered. I also noticed that the deletion methods are retrieved from a function `user_cancel_methods()`, which gives a centralised place where to alter the available methods.

I also noticed that by having a user with these two permissions above and trying to actually delete the user causes a 403, as a few routes have been disabled for non-superadmin users (while the `entity.user.cancel_form` has been left out). Thus I'm not sure if having the `delete user` permission is useful.

Reply:
I agree. 

1. The cancel methods are properly altered though as the `hook_user_cancel_methods_alter()` is being used. So this is the way to go.
2. Removed permission. We've already made the decision earlier when blocking these routes but I forgot. So it's my bad that i threw this confusion to add a permission.